src/runtime/simulator.cc

```diff
@@ -39,29 +39,841 @@ bool ParallelConfig::is_data_parallel() const
+          for (size_t i = 2; i < words.size(); i++) {
+            set_comm_path(inter_socket_gpu_fb_mem_to_gpu_fb_mem, words[i]);
+            printf("%s ", words[i].c_str());
+          }
+          printf("\n");
+        }
+        else if (words[0] == "inter_node_gpu_fb_mem_to_gpu_fb_mem") {
+          printf("inter_node_gpu_fb_mem_to_gpu_fb_mem = ");
+          for (size_t i = 2; i < words.size(); i++) {
+            set_comm_path(inter_node_gpu_fb_mem_to_gpu_fb_mem, words[i]);
+            printf("%s ", words[i].c_str());
+          }
+          printf("\n");
+        }
+        else if (words[0] == "intra_socket_sys_mem_to_gpu_fb_mem") {
+          printf("intra_socket_sys_mem_to_gpu_fb_mem = ");
+          for (size_t i = 2; i < words.size(); i++) {
+            set_comm_path(intra_socket_sys_mem_to_gpu_fb_mem, words[i]);
+            printf("%s ", words[i].c_str());
+          }
+          printf("\n");
+        }
+        else if (words[0] == "inter_socket_sys_mem_to_gpu_fb_mem") {
+          printf("inter_socket_sys_mem_to_gpu_fb_mem = ");
+          for (size_t i = 2; i < words.size(); i++) {
+            set_comm_path(inter_socket_sys_mem_to_gpu_fb_mem, words[i]);
+            printf("%s ", words[i].c_str());
+          }
+          printf("\n");
+        }
+        else if (words[0] == "inter_node_sys_mem_to_gpu_fb_mem") {
+          printf("inter_node_sys_mem_to_gpu_fb_mem = ");
+          for (size_t i = 2; i < words.size(); i++) {
+            set_comm_path(inter_node_sys_mem_to_gpu_fb_mem, words[i]);
+            printf("%s ", words[i].c_str());
+          }
+          printf("\n");
+        }
+        else if (words[0] == "intra_socket_gpu_fb_mem_to_sys_mem") {
+          printf("intra_socket_gpu_fb_mem_to_sys_mem = ");
+          for (size_t i = 2; i < words.size(); i++) {
+            set_comm_path(intra_socket_gpu_fb_mem_to_sys_mem, words[i]);
+            printf("%s ", words[i].c_str());
+          }
+          printf("\n");
+        }
+        else if (words[0] == "inter_socket_gpu_fb_mem_to_sys_mem") {
+          printf("inter_socket_gpu_fb_mem_to_sys_mem = ");
+          for (size_t i = 2; i < words.size(); i++) {
+            set_comm_path(inter_socket_gpu_fb_mem_to_sys_mem, words[i]);
+            printf("%s ", words[i].c_str());
+          }
+          printf("\n");
+        }
+        else if (words[0] == "inter_node_gpu_fb_mem_to_sys_mem") {
+          printf("inter_node_gpu_fb_mem_to_sys_mem = ");
+          for (size_t i = 2; i < words.size(); i++) {
+            set_comm_path(inter_node_gpu_fb_mem_to_sys_mem, words[i]);
+            printf("%s ", words[i].c_str());
+          }
+          printf("\n");
+        }
+      }
+    }
+  }
+  
+  num_sockets = num_nodes * num_sockets_per_node;
+  num_cpus = num_sockets * num_cpus_per_socket;
+  num_gpus = num_sockets * num_gpus_per_socket;
+  num_nvlinks_per_node = 0;
+  mem_to_nvlink.clear();
+  add_cpus();
```

Comment:
I'd personally prefer `this->` in front of these to make clear that they're methods and not functions, but this is a small thing. I defer to @jiazhihao on this

Reply:
fixed